dotCMS/src/main/java/com/dotcms/graphql/datafetcher/FieldDataFetcher.java

```diff
@@ -48,10 +55,42 @@ public Object get(final DataFetchingEnvironment environment) throws Exception {
                 }
             }
 
-            return fieldValue;
+            final boolean renderField = Try.of(()-> (boolean) environment.getArgument("render"))
+                    .getOrElse(false);
+
+            return renderField && isFieldRenderable(field)
+                    ? renderFieldValue(environment, fieldValue)
+                    : fieldValue;
         } catch (Exception e) {
             Logger.error(this, e.getMessage(), e);
             throw e;
         }
     }
+
+    private Object renderFieldValue(DataFetchingEnvironment environment, Object fieldValue) {
+        final String fieldValueAsStr = fieldValue.toString();
+
+        final HttpServletRequest request = ((DotGraphQLContext) environment.getContext())
+                .getHttpServletRequest();
+
+        final HttpServletResponse response = ((DotGraphQLContext) environment.getContext())
+                .getHttpServletResponse();
+
+        final org.apache.velocity.context.Context context = VelocityUtil
+                .getInstance().getContext(request, response);
+
```

Comment:
What else can/should we add to the velocity context here?    I think we should add at least the found contentlet object to the context as "content" and/or contentlet, like we do in the workflow processor

Reply:
makes sense @wezell